lib/rules/await-fire-event.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const { getDocsUrl } = require('../utils');
+
+function isAwaited(node) {
+  return VALID_PARENTS.includes(node.type);
+}
+
+function hasThenProperty(node) {
+  return node.type === 'MemberExpression' && node.property.name === 'then';
+}
+
+function isPromiseResolved(node) {
+  const parent = node.parent.parent;
+
+  // fireEvent.click().then(...)
+  if (parent.type === 'CallExpression') {
+    return hasThenProperty(parent.parent);
+  }
+}
```

Comment:
These methods are partly duplicated, but I prefer to leave them like that for now as they are not exactly the same from await-async-query and see in the future how to refactor them.